streams/src/main/java/org/apache/kafka/streams/kstream/internals/ChangedSerializer.java

```diff
@@ -130,7 +130,11 @@ public byte[] serialize(final String topic, final Headers headers, final Change<
             throw new StreamsException("Both old and new values are null in ChangeSerializer, which is not allowed.");
         }
 
-        return buf.array();
+        final byte[] serialized = new byte[buf.position()];
+        buf.position(0);
+        buf.get(serialized);
+
+        return serialized;
```

Comment:
then you don't need this change or any comment to explain why we need to resize to a smaller array.